util/redact/redact.go

```diff
@@ -0,0 +1,76 @@
+package redact
+
+import (
+	"encoding/hex"
+	"unsafe"
+
+	"go.uber.org/atomic"
+)
+
+// RedactLogEnabled defines whether the arguments of Error need to be redacted.
+var RedactLogEnabled atomic.String
+
+const (
+	RedactLogEnable  string = "ON"
+	RedactLogDisable string = "OFF"
+	RedactLogMarker  string = "MARKER"
+)
+
+// NeedRedact returns whether to redact log
+func NeedRedact() bool {
+	mode := RedactLogEnabled.Load()
+	return mode != RedactLogDisable && mode != ""
+}
+
+// Key receives a key return omitted information if redact log enabled
+func Key(key []byte) string {
+	if NeedRedact() {
+		return "?"
+	}
+	return String(ToUpperASCIIInplace(EncodeToString(key)))
+}
+
+// KeyBytes receives a key return omitted information if redact log enabled
+func KeyBytes(key []byte) []byte {
+	if NeedRedact() {
+		return []byte{'?'}
+	}
+	return ToUpperASCIIInplace(EncodeToString(key))
+}
+
+// String converts slice of bytes to string without copy.
+func String(b []byte) (s string) {
```

Comment:
maybe move them to `pingcap/errors`

Reply:
https://github.com/pingcap/errors/pull/75